### PR TITLE
Nn 3178 handle validation errors for case notes

### PR DESCRIPTION
**Pull request by sp-watson**
Handle prison-api validation of case notes correctly

### CODE REVIEW

File: backend/tests/caseNote.test.js [hunk 1]
```diff
@@ -187,6 +188,38 @@ describe('case note management', () => {
...
+          body: {
+            type: 'P',
+            subType: 'PI',
+            date: '20/07/2020',
+            hours: '10',
+            minutes: '10',
+            text: 'test',
+          },
+        }
+        const error400 = {
```
Comment (JamieDraperUK) [inline backend/tests/caseNote.test.js:204]: Not a massive issue but you've tested this slightly differently above.

File: backend/controllers/caseNote.js [hunk 2]
```diff
@@ -178,6 +178,30 @@ const caseNoteFactory = (prisonApi, caseNotesApi) => {
       // has been handled by the above validations
     }
 
+    if (errors.length === 0) {
```
Comment (JamieDraperUK) [inline backend/controllers/caseNote.js:181]: Might be neater pulling this out into a function

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: backend/controllers/caseNote.js (hunk 2 above)
Comment (sp-watson) [reply backend/controllers/caseNote.js:181]: I tried that, and it didn't improve readability for me due to the way it returned the error(s) or threw one.

Maybe you had a specific impl in mind that might be better?

File: backend/tests/caseNote.test.js (hunk 1 above)
Comment (sp-watson) [reply backend/tests/caseNote.test.js]: Fixed it. I did that originally and moved so I could change quicker as things weren't working

**Review by JamieDraperUK — APPROVED** [review]
LGTM :+1:

**Pull request merged**